### ISSUE TITLE
MD044: code samples in HTML comments are flagged from 'names' configuration, even if wrapped in backticks

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#462]
### rumdl version

0.1.28

### Installation method

mise

### Bug description

Similar to the bug raised in: https://github.com/rvben/rumdl/issues/457.

Content in HTML comments can be checked by MD044 (https://github.com/rvben/rumdl/blob/957ec0a374bc34a87c49653a58b821588233c4e0/docs/md044.md). This is good because it often contains prose we might want to keep to a standard.

However, unlike regular Markdown, backticks can't be used to escape the check for code samples. Ideally, what is possible in regular Markdown should be possible in HTML comments.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: Level 1 heading
   ---

   This is a Test command: `test_command`.

   <!-- This is a Test command: test_command. -->
   <!-- This is a Test command: `test_command`. -->
   ```
1. Create a `.rumdl.toml` file with this content:
   ```toml
   [MD044]
   names = ["Test"]
   ```
1. Run `rumdl check` on the Markdown file.

### Expected behavior

I would expect `rumdl` to flag a violation on line 7, because `test_command` isn't wrapped in backticks and is indistinguishable from regular prose.

I wouldn't expect `rumdl` to flag a violation on line 8, because `test_command` is wrapped in backticks and so should be treated as code and not as prose.

### Actual behavior

`rumdl` flags violations on both line 7 and on line 8.

### Configuration

```toml
[MD044]
names = ["Test"]
```

### Operating system

macOS

**Comment by rvben** [comment]
Thanks @eread for raising the issue. While this is not so much a bug, I understand the usecase and will add support for this in the upcoming release.

**Comment by eread** [comment]
> Thanks [@eread](https://github.com/eread) for raising the issue. While this is not so much a bug, I understand the usecase and will add support for this in the upcoming release.

@rvben Ah, thanks for reclassifying 🙇🏻

**Comment by rvben** [comment]
This has been resolved since v0.1.29. MD044 now respects backtick-delimited code spans inside HTML comments, so `test_command` within `<!-- -->` blocks is treated as code and no longer flagged.

Thanks @eread for the report!
